jdbcSource/src/test/java/io/vantiq/extsrc/jdbcSource/TestJDBC.java

```diff
@@ -386,6 +444,104 @@ public void testExtendedTypes() throws VantiqSQLException {
             fail("Should not have thrown exception.");
         }
     }
+
+    @Test
+    public void testParallelDates() throws VantiqSQLException {
+        assumeTrue(testDBUsername != null && testDBPassword != null && testDBURL != null && jdbcDriverLoc != null);
+        jdbc.setupJDBC(testDBURL, testDBUsername, testDBPassword, false, 0);
+        int publishResult;
+
+
+        // Create table with odd types including timestamps, dates, times, and decimals
+        try {
+            publishResult = jdbc.processPublish(CREATE_TABLE_MANY_DATES);
+            assert publishResult == 0;
+        } catch (VantiqSQLException e) {
+            e.printStackTrace();
+            fail("Should not have thrown exception." + e);
+        }
+
+        // Insert values into the newly created table
+        // Here, we will set up a bunch of sets of 25 rows, where each row has a bunch of dates in it.
+
+        ArrayList<Map<String, Object>> dataMap = new ArrayList<>();
+        try {
+            for (int i = 0; i < 25; i++) {
+                Map<String, Object> toPub = createManyDatesRows(i, PARALLEL_DATE_INSTANCE_COUNT);
+                dataMap.add(toPub);
+                for (String row : (String[]) toPub.get("batch")) {
+                    publishResult = jdbc.processPublish(row);
+                    assert publishResult > 0;
+                }
+            }
+        } catch (VantiqSQLException e) {
+            e.printStackTrace();
+            fail("Should not have thrown exception on insert: " + e);
+        }
+
+        // Now, to verify our parallel operations, we'll fire off a bunch of threads all doing the same
+        // query across this table full of dates.  We should get everything with correct results,
+        // but we had had the date results scrambled due to use of non-threadsafe methods.
+
+        AtomicReference<Exception> threadFailed = new AtomicReference<>();
+        Runnable querier = () -> {
+            // Select the values from the table and make sure the data is retrieved correctly
+            try {
+                Map[] qres = jdbc.processQuery(SELECT_QUERY_MANY_DATES);
```

Comment:
In each thread, read the entire table of dates & make sure it makes sense.  If any exceptions are thrown, save them for the master test to detect.